### PR TITLE
add instructive commnets, fix bug in collecting diagonal values

### PR DESCRIPTION
**Pull request by yairchn**
some bug fixes in corner plots

### CODE REVIEW

File: src/corner_plot.py [hunk 1]
```diff
@@ -76,25 +76,45 @@ def main():
             labelx = name_dict.get(x_matrix[i,j])
             labely = name_dict.get(y_matrix[i,j])
             if i==j:
+                #             Marginal loss on the diagonal:
```
Comment (ilopezgp) [inline src/corner_plot.py:79]:
```suggestion
                # Marginal loss on the diagonal:
```

File: src/corner_plot.py [hunk 2]
```diff
@@ -76,25 +76,45 @@ def main():
...
                 # if on the diagonal: collect all fields that has this parameter,
                 # take thier average with respect to the other parameter and plot
                 # the marginal loss curve on log-scale by scanning vertically
```
Comment (ilopezgp) [inline src/corner_plot.py:82]:
```suggestion
                # Collect all fields that have this parameter,
                # take their average with respect to the other parameter and plot
                # the marginal loss curve on log-scale by scanning vertically
```

File: src/corner_plot.py [hunk 3]
```diff
@@ -76,25 +76,45 @@ def main():
...
-                    z = np.squeeze(np.array(data.groups[group_name].variables["loss_data"])[0,0,:,:])
-                    z_diag = np.log(np.add(z_diag,np.nanmean(z, axis = 0)))
-                for k in range(i+1,M): # scan vertically
-                    group_name = z_matrix[k,j]
-                    if j==0:
-                        x = np.array(data.groups[group_name].variables[x_matrix[i,j]])
-                        z_diag = np.zeros_like(x)
-                    z = np.squeeze(np.array(data.groups[group_name].variables["loss_data"])[0,0,:,:])
-                    z_diag = np.log(np.add(z_diag, np.nanmean(z, axis = 1)))
+                # and horiozntally from this diagonal spot.
```
Comment (ilopezgp) [inline src/corner_plot.py:83]:
```suggestion
                # and horizontally from this diagonal spot.
```

File: src/corner_plot.py [hunk 4]
```diff
@@ -76,25 +76,45 @@ def main():
...
+                # Here for the variable in [1,1] collect [1,0] from the horizontal and [2,1], [3,1] from the veritcal axis
+                # for the variable in [2,2] collect [2,0] [2,1] from the horizontal and [3,2] from the veritcal axis
+                # generally for [k,k] collect from the horizontal axis [0:k-1] and from the veritcal axis [k,k+1:M]
```
Comment (ilopezgp) [inline src/corner_plot.py:86]:
```suggestion
                # Here for the variable in [1,1] collect [1,0] from the horizontal and [2,1], [3,1] from the vertical axis
                # for the variable in [2,2] collect [2,0] [2,1] from the horizontal and [3,2] from the vertical axis
                # generally for [k,k] collect from the horizontal axis [0:k-1] and from the vertical axis [k,k+1:M]
```

File: src/corner_plot.py [hunk 5]
```diff
@@ -76,25 +76,45 @@ def main():
...
+                        # if at k=0 (first instant of this variable) create z_diag
+                        if k==0:
+                            x = np.array(data.groups[group_name].variables[x_matrix[i,j]])
+                            z_diag = np.zeros_like(x)
+                        z = np.squeeze(np.array(data.groups[group_name].variables["loss_data"])[0,0,:,:])
+                        z_mean = np.nanmean(z, axis = 1)
+                        z_diag = np.add(z_diag, z_mean)
+                    elif k>i:
+                        group_name = z_matrix[k,j]
+                        # if at i=j=0 we never stepd into the previous condition create z_diag
```
Comment (ilopezgp) [inline src/corner_plot.py:108]:
```suggestion
                        # if at i=j=0 we never stepped into the previous condition create z_diag
```
?

**Review by ilopezgp — APPROVED** [review]
LGTM

**Review by haakon-e — APPROVED** [review]

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**